### PR TITLE
Upload thumbnail test & mocking

### PR DESCRIPTION
**Pull request by sisygoboom**
- create test for thumbnail uploading endpoint
- updates to the mock factory and which modules it affects in different environments:
no longer affects repositories as we're using a replica set in staging anyway
no longer affects staging: ipfs and hive modules, as we need to test these in prod-like environments

### CODE REVIEW

File: src/services/uploader/test-resources/test.png
Comment (Geo25rey) [inline src/services/uploader/test-resources/test.png:1]: No binary files. We don't need another libxz hack.

File: src/repositories/video/video.repository.mock.ts
Comment (Geo25rey) [inline src/repositories/video/video.repository.mock.ts:1]: Why is this deleted?

File: src/repositories/creator/creator.repository.mock.ts
Comment (Geo25rey) [inline src/repositories/creator/creator.repository.mock.ts:1]: And this?

**Review by Geo25rey — CHANGES_REQUESTED** [review]
Why the deleted mocks?

Absolutely no binary files unless 100% necessary.

File: src/services/uploader/test-resources/test.png
Comment (sisygoboom) [reply src/services/uploader/test-resources/test.png:1]: I'd normally agree with you, but in this case, it's 670 bytes, which is barely enough for the image data, never mind a backdoor. Additionally, it's only used in test code.

File: src/repositories/video/video.repository.mock.ts
Comment (sisygoboom) [reply src/repositories/video/video.repository.mock.ts:1]: I explain in the pr description that we use a replica set in the staging env and locally we run a mongo db via docker compose so there's actually never any need to mock the mongo calls

File: src/repositories/creator/creator.repository.mock.ts
Comment (sisygoboom) [reply src/repositories/creator/creator.repository.mock.ts:1]: same reason

File: src/services/uploader/test-resources/test.png
Comment (Geo25rey) [reply src/services/uploader/test-resources/test.png:1]: That's exactly how to xz hack happened. No binaries for tests. Maybe use an svg or something?

File: src/repositories/video/video.repository.mock.ts
Comment (Geo25rey) [reply src/repositories/video/video.repository.mock.ts:1]: I get that, but what happens when you want to do unit tests that don't rely on the db?

File: src/services/uploader/test-resources/test.png
Comment (Geo25rey) [reply src/services/uploader/test-resources/test.png:1]: You can use [this](https://www.npmjs.com/package/svg2png-wasm) if you really need a png

Comment (sisygoboom) [reply src/services/uploader/test-resources/test.png:1]: That package literally needs to execute a mystery 2mb wasm binary in order to work, I'd argue much larger security risk
image
https://github.com/ssssota/svg2png-wasm?tab=readme-ov-file#nodejs

File: src/repositories/video/video.repository.mock.ts
Comment (sisygoboom) [reply src/repositories/video/video.repository.mock.ts:1]: Very valid point, sometimes someone might need to mock out a function to perform a test if they don't want to set up the test in a way which simulates MongoDB in memory. However (and this is just my opinion) unit tests should test pure functions, if there are db calls in a function it becomes an integration test. in which case it should be verified the items are being correctly inserted into in-memory db

File: src/services/uploader/test-resources/test.png
Comment (sisygoboom) [reply src/services/uploader/test-resources/test.png:1]: I'd also like to add that it seems the svgs in their codebase are actually base64 encoded pngs with SVG wrappers https://github.com/ssssota/svg2png-wasm/blob/main/site/static/parrot.svg?short_path=4ff033b

Comment (Geo25rey) [reply src/services/uploader/test-resources/test.png:1]: Here's a pure JS implementation https://www.npmjs.com/package/@michaelheerklotz/svgexport

File: src/repositories/video/video.repository.mock.ts
Comment (Geo25rey) [reply src/repositories/video/video.repository.mock.ts:1]: I don't think it hurts to leave it in.

**Review by Geo25rey — APPROVED** [review]
LGTM

**Pull request merged**